src/toolkit/matl_sell_policy.h

```diff
@@ -0,0 +1,139 @@
+#ifndef CYCLUS_SRC_TOOLKIT_MATL_SELL_POLICY_H_
+#define CYCLUS_SRC_TOOLKIT_MATL_SELL_POLICY_H_
+
+#include <string>
+
+#include "composition.h"
+#include "material.h"
+#include "resource_buff.h"
+#include "trader.h"
+
+namespace cyclus {
+namespace toolkit {
+
+/// MatlSellPolicy performs semi-automatic inventory management of a material
+/// buffer by making offers and trading away materials in an attempt to empty
+/// the buffer's inventory every time step.
+///
+/// For simple behavior, policies virtually eliminate the need to write any code
+/// for resource exchange. Just assign a few policies to work with a few buffers
+/// and focus on writing the physics and other behvavior of your agent.  Typical
+/// usage goes something like this:
+///
+/// @code
+/// class YourAgent : public Facility {
+///  public:
+///   ...
+///
+///   void EnterNotify() {
+///     Facility::EnterNotify(); // always do this first
+///
+///     policy_.Init(this, &outbuf_, "outbuf-label").Set(outcommod).Start();
+///   }
+///   ...
+///
+///  private:
+///   MatlSellPolicy policy_;
+///   ResourceBuff outbuf_;
+///    ...
+/// }
+/// @endcode
+///
+/// The policy needs to be initialized with its owning agent and the material
+/// buffer that is is managing. It also needs to be activated by calling the
+/// Start function for it to begin participation in resource exchange.  And
+/// don't forget to add some commodities to offer on by calling Set.  All policy
+/// configuration should usually occur in the agent's EnterNotify member
+/// function.
+///
+/// @warn When a policy's managing agent is deallocated, you MUST either
+/// call the policy's Stop function or delete the policy. Otherwise SEGFAULT.
+class MatlSellPolicy : public Trader {
+ public:
+  /// Creates an uninitialized policy.  The Init function MUST be called before
+  /// anything else is done with the policy.
+  MatlSellPolicy();
+
+  virtual ~MatlSellPolicy();
+
+  /// Configures the policy to keep buf empty every time step by offering away
+  /// as much material as possible.  If quantize is greater than zero, the
+  /// policy will make exclusive, integral quantize kg offers.  Otherwise, a
+  /// single offer will be sent each time step to empty the buffer's entire
+  /// inventory.
+  /// @param manager the agent
+  /// @param buf the resource buffer
+  /// @param name a unique name identifying this policy
+  /// @param throughput a maximum throughput per time step for material leaving
+  /// buf
+  /// @ignore_comp ignore the composition of material in buf. When making bids,
+  /// requested material compsositions will be used. When making trades,
+  /// material in the buffer will have its composition transmuted to match the
+  /// requested material.
```

Comment:
I'm confused by this - is the requested/offered mismatch option in case material needs to decay over time?  Shouldn't the 'buying' facility just accept the different material or reject the offer?